pkg/backend/filestate/gcpauth.go

```diff
@@ -0,0 +1,77 @@
+package filestate
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+
+	"github.com/pulumi/pulumi/pkg/diag"
+	"github.com/pulumi/pulumi/pkg/util/cmdutil"
+
+	"golang.org/x/oauth2/google"
+
+	"gocloud.dev/blob/gcsblob"
+
+	"cloud.google.com/go/storage"
+
+	"github.com/pkg/errors"
+	"gocloud.dev/blob"
+	"gocloud.dev/gcp"
+)
+
+type GoogleCredentials struct {
+	PrivateKeyID string `json:"private_key_id"`
+	PrivateKey   string `json:"private_key"`
+	ClientEmail  string `json:"client_email"`
+	ClientID     string `json:"client_id"`
+}
+
+func GoogleCredentialsMux() (*blob.URLMux, error) {
+
+	var err error
+	var credentials *google.Credentials
+
+	if creds := os.Getenv("GOOGLE_CREDENTIALS"); creds != "" {
+		// We try $GOOGLE_CREDENTIALS before gcp.DefaultCredentials
+		// so that users can override the default creds
```

Comment:
can we link to something like https://www.terraform.io/docs/backends/types/gcs.html to explain where GOOGLE_CREDENTIALS came from?

Reply:
done